src/iotconnect_common.c

```diff
@@ -9,18 +9,17 @@
 
 static char timebuf[sizeof "2011-10-08T07:07:01.000Z"];
 
-static const char *to_iso_timestamp(time_t *timestamp) {
-    time_t ts = timestamp ? *timestamp : time(NULL);
+// Obviously, this function is not thread-safe, due to the fixed buffer
+const char *iotcl_iso_timestamp_now(void) {
+    time_t ts = time(NULL);
     strftime(timebuf, (sizeof timebuf), "%Y-%m-%dT%H:%M:%S.000Z", gmtime(&ts));
     return timebuf;
 }
 
-const char *iotcl_to_iso_timestamp(time_t timestamp) {
-    return to_iso_timestamp(&timestamp);
-}
-
-const char *iotcl_iso_timestamp_now(void) {
-    return to_iso_timestamp(NULL);
+unsigned long get_expiry_from_now(unsigned long int expiry_secs)
```

Comment:
I think we should use time_t in the argument list and return value for general compatibility

Reply:
I had terrible trouble using time_t on DA-16200 -- there was some awful issue with casting.
I'll have another look, in case a different approach works -- maybe it can be finessed.

Reply:
Maybe you had issues printing it? It happened to me on several occasions, but all I did there was to cast time_t to unsigned long if I was printf-ing %lu for example.

It doesn't change the fact that time diff is still of type time_t as as is the return of the system time() per C standard specification for time() function. This should apply to the return value type and expiry_secs argument. The function should still return time_t and whatever issues are present in DA should be addressed at DA level.